### PR TITLE
[CELEBORN-843][BUILD] `sbt` support flink-related module build/test

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

This PR adds packaging and testing support for Flink-related modules using SBT based on #1757 

### Why are the changes needed?

improve project build speed

running flink-it tests with -Pflink-1.14

```shell
sbt:celeborn> project flink-it
sbt:flink-it> clean
sbt:flink-it> test
[success] Total time: 136 s (02:16), completed 2023-7-27 11:55:10
```

running flink-it tests with -Pflink-1.17

```shell
$ ./build/sbt -Pflink-1.17
sbt:celeborn> project flink-it
sbt:flink-it> clean
sbt:flink-it> test
[success] Total time: 168 s (02:48), completed 2023-7-27 11:28:35
```

packing and shading the flink 1.14 client

```shell
$ ./build/sbt -Pflink-1.14
sbt:celeborn> clean
sbt:celeborn> project celeborn-client-flink-1_14-shaded
sbt:celeborn-client-flink-1_14-shaded> assembly
[success] Total time: 35 s, completed 2023-7-27 11:51:54
```

packing and shading the flink 1.17 client

```shell
$ ./build/sbt -Pflink-1.17
sbt:celeborn> clean
sbt:celeborn> project celeborn-client-flink-1_17-shaded
sbt:celeborn-client-flink-1_17-shaded> assembly
[success] Total time: 39 s, completed 2023-7-27 11:49:20
```

### Does this PR introduce _any_ user-facing change?

yes

### How was this patch tested?

tested locally

### CODE REVIEW

**Review by pan3793 — APPROVED** [review]

**Comment by cfmcgrady** [comment]
diff between the `celeborn-client-flink-1.17-shaded_2.12-0.4.0-SNAPSHOT.jar` file in sbt and maven:

```diff
diff --git a/maven-files.out b/sbt-files.out
index 8228cae..8db58f9 100644
--- a/maven-files.out
+++ b/sbt-files.out
@@ -1,19 +1,6 @@
 ./celeborn-client-flink-1.17-shaded_2.12-0.4.0-SNAPSHOT.jar
-./google/protobuf/any.proto
-./google/protobuf/api.proto
-./google/protobuf/compiler/plugin.proto
-./google/protobuf/descriptor.proto
-./google/protobuf/duration.proto
-./google/protobuf/empty.proto
-./google/protobuf/field_mask.proto
-./google/protobuf/source_context.proto
-./google/protobuf/struct.proto
-./google/protobuf/timestamp.proto
-./google/protobuf/type.proto
-./google/protobuf/wrappers.proto
-./META-INF/DEPENDENCIES
+./META-INF/INDEX.LIST
 ./META-INF/io.netty.versions.properties
-./META-INF/LICENSE
 ./META-INF/LICENSE.txt
 ./META-INF/MANIFEST.MF
 ./META-INF/maven/com.google.guava/guava/pom.properties
@@ -80,53 +67,12 @@
 ./META-INF/maven/io.netty/netty-transport-sctp/pom.xml
 ./META-INF/maven/io.netty/netty-transport-udt/pom.properties
 ./META-INF/maven/io.netty/netty-transport-udt/pom.xml
-./META-INF/maven/org.apache.celeborn/celeborn-client_2.12/pom.properties
-./META-INF/maven/org.apache.celeborn/celeborn-client_2.12/pom.xml
-./META-INF/maven/org.apache.celeborn/celeborn-client-flink-1.17_2.12/pom.properties
-./META-INF/maven/org.apache.celeborn/celeborn-client-flink-1.17_2.12/pom.xml
-./META-INF/maven/org.apache.celeborn/celeborn-client-flink-1.17-shaded_2.12/pom.properties
-./META-INF/maven/org.apache.celeborn/celeborn-client-flink-1.17-shaded_2.12/pom.xml
-./META-INF/maven/org.apache.celeborn/celeborn-client-flink-common_2.12/pom.properties
-./META-INF/maven/org.apache.celeborn/celeborn-client-flink-common_2.12/pom.xml
-./META-INF/maven/org.apache.celeborn/celeborn-common_2.12/pom.properties
-./META-INF/maven/org.apache.celeborn/celeborn-common_2.12/pom.xml
 ./META-INF/maven/org.apache.commons/commons-lang3/pom.properties
 ./META-INF/maven/org.apache.commons/commons-lang3/pom.xml
 ./META-INF/maven/org.jctools/jctools-core/pom.properties
 ./META-INF/maven/org.jctools/jctools-core/pom.xml
-./META-INF/native-image/io.netty/netty-buffer/native-image.properties
-./META-INF/native-image/io.netty/netty-codec-dns/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec-haproxy/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec-http2/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec-http2/native-image.properties
-./META-INF/native-image/io.netty/netty-codec-http/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec-http/native-image.properties
-./META-INF/native-image/io.netty/netty-codec-memcache/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec-mqtt/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec/native-image.properties
-./META-INF/native-image/io.netty/netty-codec-redis/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec-smtp/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec-socks/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec-stomp/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-codec-xml/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-common/native-image.properties
-./META-INF/native-image/io.netty/netty-handler/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-handler/native-image.properties
-./META-INF/native-image/io.netty/netty-handler-proxy/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-resolver-dns/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-resolver-dns/native-image.properties
-./META-INF/native-image/io.netty/netty-transport-classes-epoll/jni-config.json
-./META-INF/native-image/io.netty/netty-transport-classes-epoll/native-image.properties
-./META-INF/native-image/io.netty/netty-transport-classes-epoll/reflect-config.json
-./META-INF/native-image/io.netty/netty-transport-classes-epoll/resource-config.json
-./META-INF/native-image/io.netty/netty-transport/generated/handlers/reflect-config.json
-./META-INF/native-image/io.netty/netty-transport/native-image.properties
-./META-INF/native-image/io.netty/netty-transport/reflection-config.json
-./META-INF/native-image/io.netty/netty-transport-sctp/generated/handlers/reflect-config.json
 ./META-INF/native/liborg_apache_celeborn_shaded_netty_transport_native_epoll_aarch_64.so
 ./META-INF/native/liborg_apache_celeborn_shaded_netty_transport_native_epoll_x86_64.so
-./META-INF/NOTICE
 ./META-INF/NOTICE.txt
 ./META-INF/services/reactor.blockhound.integration.BlockHoundIntegration
 ./org/apache/celeborn/client/ApplicationHeartbeater$$anon$1.class
@@ -844,6 +790,8 @@
 ./org/apache/celeborn/common/write/SimplePushStrategy.class
 ./org/apache/celeborn/common/write/SlowStartPushStrategy$CongestControlContext.class
 ./org/apache/celeborn/common/write/SlowStartPushStrategy.class
+./org/apache/celeborn/package$.class
+./org/apache/celeborn/package.class
 ./org/apache/celeborn/plugin/flink/AbstractRemoteShuffleEnvironment.class
 ./org/apache/celeborn/plugin/flink/AbstractRemoteShuffleInputGateFactory.class
 ./org/apache/celeborn/plugin/flink/AbstractRemoteShuffleResultPartitionFactory.class
@@ -6664,4 +6612,3 @@
 ./org/apache/celeborn/shaded/org/roaringbitmap/RunContainer.class
 ./org/apache/celeborn/shaded/org/roaringbitmap/Util.class
 ./org/apache/celeborn/shaded/org/roaringbitmap/WordStorage.class
-./TransportMessages.proto
```

**Comment by waitinfuture** [comment]
Ping @RexXiong @FMX @zhongqiangczq could you take a look at this PR? thanks!

File: project/CelebornBuild.scala [hunk 1]
```diff
@@ -202,7 +213,7 @@ object CelebornCommon {
         "io.dropwizard.metrics" % "metrics-core" % metricsVersion,
         "io.dropwizard.metrics" % "metrics-graphite" % metricsVersion,
         "io.dropwizard.metrics" % "metrics-jvm" % metricsVersion,
-        "io.netty" % "netty-all" % nettyVersion,
+        ("io.netty" % "netty-all" % nettyVersion).exclude("io.netty", "netty-handler-ssl-ocsp"),
```
Comment (pan3793) [inline project/CelebornBuild.scala:216]: can you remove this change? as we discussed before, do exclusion after introducing the dependency list audit mechanism.

Comment (cfmcgrady) [reply project/CelebornBuild.scala:216]: updated

**Comment by zhongqiangczq** [comment]
can sbt still use the maven local repository to avoid download dependencies again ?

**Comment by cfmcgrady** [comment]
> can sbt still use the maven local repository to avoid download dependencies again ?

Good catch, it should be, let me raise a separate PR to support this.

**Review by zhongqiangczq — APPROVED** [review]
LGTM

**Comment by pan3793** [comment]
Thanks, merged to main

**Pull request closed without merging**